### ISSUE TITLE
CTE removal (unnecessary generated CTEs)

### ISSUE THREAD

**Issue by philpep** [PRQL/prql#2911]
### What happened?

Hi,

I noticed some case of unnecessary generated CTE using `let`.  I think this can be avoided when CTE is used only once and some other - to be determined - conditions.

This might not be considered as a bug as prql input could be written in a way that it doesn't generate a CTE but it may be useful to think about prql or AST rewriting feature in the compiler ?

Thanks!

### PRQL input

```elm
let t = (from r)
from t
```

### SQL output

```SQL
WITH t AS (
  SELECT
    *
  FROM
    r
)
SELECT
  *
FROM
  t
```

### Expected SQL output

```SQL
SELECT
  *
FROM
  r
```

### MVCE confirmation

- [X] Minimal example
- [X] New issue

**Comment by max-sixty** [comment]
Yes, I thought we had an issue on this, but can't find it and am probably misremembering.

We could indeed reduce these. In the past I've seen examples where it's less clear-cut — it's possible to reduce, but changing a parameter slightly means it's helpful to have them separate. One option would be to do a pass of the final SQL AST and collapse any unnecessary CTEs — that would mean the earlier compilation stages could remain unconcerned about being concise.

(I'll switch from Bug to Friendliness, thanks for filling out the template though @philpep )
